src/services/pseudonym/hooks/index.js

```diff
@@ -42,14 +42,19 @@ const filterValidUsers = (context) => {
 	let validUserIds = [context.params.account.userId];
 	return context.app.service('courses').find({
 		query: {
-			teacherIds: context.params.account.userId,
+			$or: [
+				{ teacherIds: context.params.account.userId },
+				{ userIds: context.params.account.userId },
+			],
 			$populate: 'classIds',
 		},
 	}).then((courses) => {
 		for (const course of courses.data) {
 			validUserIds = validUserIds.concat(course.userIds);
+			validUserIds = validUserIds.concat(course.teacherIds);
```

Comment:
this could be done shorter than multiple reassigns of `validUserIds` like https://www.samanthaming.com/tidbits/49-2-ways-to-merge-arrays